From 3afe3725fcf21fab7204243b9485a118e499b4a3 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 25 May 2011 13:09:08 +0200
Subject: [PATCH] readahead-collect: ignore EACCES for fanotify

At the start of auditd, we are temporarily not able to read
from the fanotify fd. Ignoring it, seems to work.
---
 src/readahead-collect.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/readahead-collect.c b/src/readahead-collect.c
index 3c48a02..913a340 100644
--- a/src/readahead-collect.c
+++ b/src/readahead-collect.c
@@ -380,7 +380,7 @@ static int collect(const char *root) {
 
                 if ((n = read(fanotify_fd, &data, sizeof(data))) < 0) {
 
-                        if (errno == EINTR || errno == EAGAIN)
+                        if (errno == EINTR || errno == EAGAIN || errno == EACCES)
                                 continue;
 
                         log_error("Failed to read event: %m");
-- 
1.7.5.2

